test/OffsetHelper.test.ts

```diff
@@ -3,29 +3,35 @@ import { expect } from "chai";
 import { ethers } from "hardhat";
 import { formatEther, parseEther, parseUnits } from "ethers/lib/utils";
 
-import * as hardhatContracts from "../utils/toucanContracts.json";
-import * as poolContract from "../artifacts/contracts/interfaces/IToucanPoolToken.sol/IToucanPoolToken.json";
 import {
+  IERC20,
+  IERC20__factory,
+  IWETH,
+  IWETH__factory,
   IToucanPoolToken,
+  IToucanPoolToken__factory,
   OffsetHelper,
   OffsetHelper__factory,
   Swapper,
   Swapper__factory,
 } from "../typechain";
 import addresses from "../utils/addresses";
-import { BigNumber, Contract } from "ethers";
-import { usdcABI, wethABI, wmaticABI } from "../utils/ABIs";
+import { BigNumber } from "ethers";
 
 const ONE_ETHER = parseEther("1.0");
 
-describe("Offset Helper - autoOffset", function () {
+function parseUSDC(s: string): BigNumber {
+  return parseUnits(s, 6);
+}
+
+describe("OffsetHelper", function () {
   let offsetHelper: OffsetHelper;
   let swapper: Swapper;
   let bct: IToucanPoolToken;
   let nct: IToucanPoolToken;
-  let weth: Contract;
-  let wmatic: Contract;
-  let usdc: Contract;
+  let weth: IERC20;
+  let wmatic: IWETH;
```

Comment:
It's interesting that you chose to add the `IWETH` type to `wmatic` and `IERC20` to the others. Why is that? 🤔

Reply:
Because `wmatic` is the actual `WETH` (wrapped "ether", that is, native currency) of polygon and used like this. E.g. only on `wmatic` is `deposit` called. The other tokens, including `weth` are just accessed as plain `IERC20`s. Note that the `IWETH` interface is a simple extension of the `IERC20` interface, so if you don't access any function of the extended interface, there's no point in using it for a token, especially if that token's contract actually doesn't even have the `deposit` and `withdrawal` methods.

Reply:
Nah, you're right. Let's keep it as is.